providers/common/include/prov/fipsindicator.h

```diff
@@ -9,83 +9,134 @@
 
 #ifdef FIPS_MODULE
 
-# define OSSL_FIPS_INDICATOR_ID_STRICT_CHECK0 0
-# define OSSL_FIPS_INDICATOR_ID_STRICT_CHECK1 1
-# define OSSL_FIPS_INDICATOR_ID_STRICT_CHECK2 2
+/*
+ * There is a single gettable which determines if an algorithm is approved, but
+ * there may be multiple flags required to keep track of the approved status.
+ * There is an id associated with each of these flags.
+ */
+# define OSSL_FIPS_IND_APPROVED_0 0
+# define OSSL_FIPS_IND_APPROVED_1 1
+# define OSSL_FIPS_IND_APPROVED_2 2
+
+/*
+ * There may be multiple settables associated with an algorithm that allow
+ * overriding the default status.
+ * We associate an id with each of these.
+ */
+# define OSSL_FIPS_IND_SETTABLE0 0
+# define OSSL_FIPS_IND_SETTABLE1 1
+# define OSSL_FIPS_IND_SETTABLE2 2
```

Comment:
Why not make the two sets of flags the same?
This is creating unnecessary duplication IMO>

Reply:
because they are completely different entities.
The settables relate to what fips configurable item we are overriding per ctx. 

The allowed id's relate to flags that make up the approved gettable.

Reply:
I would have thought that the bits you set would correspond to the bits you get.

Since these _aren't_ public symbols, it's not a big deal.

Reply:
what is the purpose of `OSSL_FIPS_IND_SETTABLE1` and `OSSL_FIPS_IND_SETTABLE2` it seems unused currently. What is the plan for those?

Reply:
If there is more than one test inside an algorithm, each check may have a different setter, so that they can be overridden separately. This PR just targets existing checks, it does not deal with any of the newer checks that will be added,